modules/admin/admin-ui/src/main/resources/web/admin/apps/content-studio/js/app/wizard/ContentWizardPanel.ts

```diff
@@ -608,6 +608,17 @@ export class ContentWizardPanel extends api.app.wizard.WizardPanel<Content> {
 
             this.setSteps(steps);
 
+            // Content can be moved from its Site, but persisted content can still have metadata
+            // Such content needs update
+            const updateHandler = () => {
+                const needUpdate = !this.getPersistedItem().extraDataEquals(this.getExtraDataFromMetadata());
+                if (needUpdate) {
+                    this.saveChanges();
+                }
```

Comment:
I agree @alansemenov. I think we should do this in the content-service.

1) Move the stuff from content-service into a new MoveContentCommand
2) When doing the move, check if its out of site
3) If out of site, do update for content and all children